### ISSUE TITLE
(react-hooks): `exports` property in `package.json` does not correctly specify `types`

### ISSUE THREAD

**Issue by PSoltes** [trycourier/courier-react#571]
```
"exports": {
    ".": "./dist/index.js",
    "./use-inbox": "./dist/inbox/use-inbox.js"
  },
```
above specification will result in ts complaining that `Could not find a declaration file for module '@trycourier/react-hooks'`. Declaring something like below solves the issue.
```
"exports": {
    ".": {
      "import": {
        "types": "./typings/index.d.ts",
        "default": "./dist/index.js"
      },
      "require": {
        "types": "./typings/index.d.ts",
        "default": "./dist/index.js"
      }
    },
    "./use-inbox": "./dist/inbox/use-inbox.js"
  },
```

**Comment by PSoltes** [comment]
node - 20.13.1
@trycourier/react-hooks - 5.1.1
typescript - 5.3.3

one more thing I think is relevant is tsconfig, esp. moduleResolution value
```
"module": "ESNext",
"target": "ESNext",
"moduleResolution": "Bundler",
```

**Comment by rileylnapier** [comment]
ok thanks, ill try this out.  ill most likely need to update a bunch of our packages if this is required for a setup like yours

**Comment by PSoltes** [comment]
Thanks. And to reproduce it, I am quite sure now, that `moduleResolution` of `bundler` or `node16/nodenext` is necessary, cause only those actually use `exports` field in `package.json`

**Comment by PSoltes** [comment]
any updates on this? should we create a PR for this thing?

**Comment by rileylnapier** [comment]
Hi!  Yep I merged the change but we need to do a release and it's bundled in a 6.0 breaking change

It should be released soon :)

**Comment by rileylnapier** [comment]
released 6.0

**Comment by PSoltes** [comment]
Thanks so much
